### ISSUE TITLE
Bad handling of lack of extended attributes leads to SFTP out of memory error

### ISSUE THREAD

**Issue by cfiehe** [proftpd/proftpd#1785]
### What I Did

The issue is really hard to reproduce. Currently I am not able to describe what actually caused the issue. I can only describe under which circumstances the out of memory exception occurs. It seems to be a very strange combination between mod_sftp, mod_vroot and SSH key order.

I must mount a share via SSHFS. All users are chrooted to their home directory. The user's home directory contains the following authorized keys:
```
---- BEGIN SSH2 PUBLIC KEY ----
Comment: "256-bit ED25519, root@myhost.my-company.com"
AAAAC3NzaC1lZDI1NTE5AAAAIObz9MKl1N4XFZCa1xJ7YGNIZm05pWD8OneUAOEQlJl2
---- END SSH2 PUBLIC KEY ----
---- BEGIN SSH2 PUBLIC KEY ----
Comment: "2048-bit RSA, root@myhost.my-company.com"
AAAAB3NzaC1yc2EAAAADAQABAAABAQDE0ZQ/cXNet/teK9fR0cjOyMKyHCbqFReuN+nhBo
svLJlTgMkEkTrfoFyibLScDY5KE1gRniMTD2PRtZ7FX/8SWeZL443TiZ8MFi27JM9KSGkx
gnctZR+95bvypzN8wcijF1Y723QUruPRGFbm/yO641Hxgd3QTBHi6GNq5wJtTTMrM9j1/7
SbcSZte3RyKf4mlMEemChonbshELQtMkcSlqChN6ihf67yBXEIJe5mRIMIohtne3AGvtFr
WC3bbDec735fajQjfpVG21tvdH7nTcUdAj4XXBiGh/oY0ZZ/+cloBsDjux9NOA8jPtazFz
I1LQi5NItUSldNe/dRwwRP
---- END SSH2 PUBLIC KEY ----
```
I can mount the share when I use the first key of type ED25519. When the second key is used, the mount hangs and fails after some seconds. The fstab file contains:
```
myuser@secdata-t.mycompany.com:/    /mnt/secdata/myuser    fuse.sshfs    noauto,x-systemd.automount,nofail,allow_other,reconnect,ServerAliveInterval=15,_netdev,IdentityFile=/var/tmp/id_rsa    0    0
```
The server log gives the following:
```
2024-03-19 17:24:51,801 9594edfb0019 proftpd[12] 0.0.0.0 (A.B.C.D[A.B.C.D]): Out of memory!
```
The user can login successfully. In htop I can see that the memory consumption of the process which executes `LSTAT` grows quickly until it reaches the memory limit. The issue is reproducible.

The issue is gone when you deactivate the VRootEngine or when you change the key order(!) in the authorized keys file by making the RSA the first key in the file.

This is a log except of the trace.log:
```
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system chroot() for path '/srv/ftp/myuser'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system lstat() for path '/'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system chdir() for path '/'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system lstat() for path '/'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system stat() for path '/'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system stat() for path '/'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system stat() for path '/.ftpaccess'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system stat() for path '/.ftpaccess'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system stat() for path '/'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system stat() for path '/'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system stat() for path '/.ftpaccess'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system stat() for path '/.ftpaccess'
2024-03-19 17:24:44,931 [12] <fsio:8>: using vroot+facl+system lstat() for path '/'
2024-03-19 17:24:44,932 [12] <ssh2:3>: sent SSH_MSG_USER_AUTH_SUCCESS (52) packet (28 bytes)
2024-03-19 17:24:44,932 [12] <fsio:8>: using vroot+facl+system stat() for path '/'
2024-03-19 17:24:44,932 [12] <fsio:8>: using vroot+facl+system stat() for path '/'
2024-03-19 17:24:44,932 [12] <fsio:8>: using vroot+facl+system stat() for path '/.ftpaccess'
2024-03-19 17:24:44,932 [12] <fsio:8>: using vroot+facl+system stat() for path '/.ftpaccess'
2024-03-19 17:24:44,932 [12] <ssh2:3>: sent SSH_MSG_GLOBAL_REQUEST (80) packet (396 bytes)
2024-03-19 17:24:44,932 [12] <ssh2:3>: received SSH_MSG_CHANNEL_OPEN (90) packet
2024-03-19 17:24:44,932 [12] <ssh2:8>: open of 'session' channel using remote ID 0 requested: initial client window len = 2097152 bytes, client max packet size = 32768 bytes
2024-03-19 17:24:44,932 [12] <ssh2:8>: confirm open channel remote ID 0, local ID 0: initial server window len = 4294967295 bytes, server max packet size = 32768 bytes
2024-03-19 17:24:44,932 [12] <ssh2:3>: sent SSH_MSG_CHANNEL_OPEN_CONFIRMATION (91) packet (44 bytes)
2024-03-19 17:24:44,933 [12] <ssh2:3>: received SSH_MSG_CHANNEL_REQUEST (98) packet
2024-03-19 17:24:44,933 [12] <ssh2:7>: received 'env' request for channel ID 0, want reply = false
2024-03-19 17:24:44,933 [12] <ssh2:3>: received SSH_MSG_CHANNEL_REQUEST (98) packet
2024-03-19 17:24:44,933 [12] <ssh2:7>: received 'subsystem' request for channel ID 0, want reply = true
2024-03-19 17:24:44,933 [12] <ssh2:3>: sent SSH_MSG_CHANNEL_SUCCESS (99) packet (36 bytes)
2024-03-19 17:24:44,933 [12] <ssh2:3>: received SSH_MSG_CHANNEL_DATA (94) packet
2024-03-19 17:24:44,933 [12] <sftp:9>: reading SFTP data from SSH2 packet buffer (9 bytes)
2024-03-19 17:24:44,933 [12] <sftp:6>: received INIT (1) SFTP request (channel ID 0)
2024-03-19 17:24:44,933 [12] <sftp:7>: received request: INIT 3
2024-03-19 17:24:44,933 [12] <sftp:8>: sending response: VERSION 3
2024-03-19 17:24:44,933 [12] <ssh2:9>: sending CHANNEL_DATA (remote channel ID 0, 302 data bytes)
2024-03-19 17:24:44,933 [12] <ssh2:3>: sent SSH_MSG_CHANNEL_DATA (94) packet (340 bytes)
2024-03-19 17:24:44,933 [12] <ssh2:3>: received SSH_MSG_CHANNEL_DATA (94) packet
2024-03-19 17:24:44,933 [12] <sftp:9>: reading SFTP data from SSH2 packet buffer (14 bytes)
2024-03-19 17:24:44,933 [12] <sftp:6>: received LSTAT (7) SFTP request (channel ID 0)
2024-03-19 17:24:44,933 [12] <sftp:7>: received request: LSTAT /
2024-03-19 17:24:44,933 [12] <fsio:8>: using vroot+facl+system lstat() for path '/'
2024-03-19 17:24:44,933 [12] <fsio:8>: using vroot+facl+system stat() for path '/'
2024-03-19 17:24:44,933 [12] <fsio:8>: using vroot+facl+system stat() for path '/'
2024-03-19 17:24:44,933 [12] <fsio:8>: using vroot+facl+system stat() for path '/.ftpaccess'
2024-03-19 17:24:44,933 [12] <fsio:8>: using vroot+facl+system stat() for path '/.ftpaccess'
2024-03-19 17:24:44,933 [12] <fsio:8>: using vroot+facl+system lstat() for path '/'
2024-03-19 17:24:44,933 [12] <fsio:8>: using vroot+facl+system lstat() for path '/'
2024-03-19 17:24:44,933 [12] <sftp:8>: sending response: ATTRS type=dir;size=18;UNIX.owner=1844400482;UNIX.group=1844400043;UNIX.mode=0700;access=20240319161152;modify=20240319161152;
2024-03-19 17:24:44,933 [12] <fsio:8>: using system llistxattr() for path '/'
```
And nearly 7 seconds after the last log entry, the memory limit is reached. By the way, the user's home directory does not contain any files.

### ProFTPD Version and Configuration

```bash
Compile-time Settings:
  Version: 1.3.9rc3 (git)
  Platform: LINUX [Linux 5.15.0-100-generic x86_64]
  OS/Release:
    PRETTY_NAME="Ubuntu 22.04.4 LTS"
    NAME="Ubuntu"
    VERSION_ID="22.04"
    VERSION="22.04.4 LTS (Jammy Jellyfish)"
    VERSION_CODENAME=jammy
    ID=ubuntu
    ID_LIKE=debian
    UBUNTU_CODENAME=jammy
  Built: Mon Mar 18 2024 12:26:35 UTC
  Built With:
    configure  'CFLAGS=-DBAN_LIST_MAXSZ=1024' '--prefix=/usr' '--mandir=/usr/share/man' '--sysconfdir=/etc/proftpd' '--localstatedir=/run' '--libexecdir=/usr/lib/proftpd' '--with-includes=/usr/include/mysql:/usr/include/postgresql' '--with-shared=mod_auth_otp:mod_ban:mod_clamav:mod_copy:mod_ctrls_admin:mod_deflate:mod_digest:mod_dnsbl:mod_dynmasq:mod_exec:mod_facl:mod_geoip:mod_ident:mod_ifsession:mod_ifversion:mod_ldap:mod_load:mod_log_forensic:mod_proxy_protocol:mod_qos:mod_quotatab:mod_quotatab_file:mod_quotatab_ldap:mod_quotatab_radius:mod_quotatab_sql:mod_radius:mod_ratio:mod_readme:mod_redis:mod_rewrite:mod_sftp:mod_sftp_pam:mod_sftp_sql:mod_shaper:mod_site_misc:mod_snmp:mod_sql:mod_sql_mysql:mod_sql_odbc:mod_sql_passwd:mod_sql_postgres:mod_statcache:mod_tls:mod_tls_fscache:mod_tls_redis:mod_tls_shmcache:mod_unique_id:mod_vroot:mod_wrap:mod_wrap2:mod_wrap2_file:mod_wrap2_redis:mod_wrap2_sql' '--enable-autoshadow' '--enable-ctrls' '--enable-dso' '--enable-facl' '--enable-ipv6' '--enable-openssl' '--enable-pcre2' '--enable-nls' '--enable-redis' '--enable-sendfile' '--with-lastlog=/var/log/lastlog' '--disable-strip' '--disable-ident'

  CFLAGS: -g2 -DBAN_LIST_MAXSZ=1024 -Wall -fno-omit-frame-pointer -fno-strict-aliasing -Werror=implicit-function-declaration
  LDFLAGS: -Wl,-L$(top_srcdir)/lib,-L$(top_builddir)/lib  -rdynamic  -L/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu
  LIBS: -lacl  -lpcre2-posix -lpcre2-8 -lssl -lcrypto -lsodium -lcap  -lpam -lattr -lidn2 -lcrypt -lhiredis  -pthread

  Files:
    Configuration File:
      /etc/proftpd/proftpd.conf
    Pid File:
      /run/proftpd.pid
    Scoreboard File:
      /run/proftpd.scoreboard
    Header Directory:
      /usr/include/proftpd
    Shared Module Directory:
      /usr/lib/proftpd

  Info:
    + Max supported UID: 4294967295
    + Max supported GID: 4294967295

  Features:
    + Autoshadow support
    + Controls support
    + curses support
    - Developer support
    + DSO support
    + IPv6 support
    + Largefile support
    + Lastlog support
    - Memcache support
    + ncursesw support
    + NLS support
    + OpenSSL support (OpenSSL 3.0.2 15 Mar 2022)
    - PCRE support
    + PCRE2 support
    + POSIX ACL support
    + Redis support
    + Sendfile support
    + Shadow file support
    + Sodium support
    + Trace support
    + xattr support

  Tunable Options:
    PR_TUNABLE_BUFFER_SIZE = 1024
    PR_TUNABLE_DEFAULT_RCVBUFSZ = 65536
    PR_TUNABLE_DEFAULT_SNDBUFSZ = 65536
    PR_TUNABLE_ENV_MAX = 2048
    PR_TUNABLE_GLOBBING_MAX_MATCHES = 100000
    PR_TUNABLE_GLOBBING_MAX_RECURSION = 8
    PR_TUNABLE_HASH_TABLE_SIZE = 40
    PR_TUNABLE_LOGIN_MAX = 256
    PR_TUNABLE_NEW_POOL_SIZE = 512
    PR_TUNABLE_PATH_MAX = 4096
    PR_TUNABLE_SCOREBOARD_BUFFER_SIZE = 80
    PR_TUNABLE_SCOREBOARD_SCRUB_TIMER = 30
    PR_TUNABLE_SELECT_TIMEOUT = 30
    PR_TUNABLE_TIMEOUTIDENT = 10
    PR_TUNABLE_TIMEOUTIDLE = 600
    PR_TUNABLE_TIMEOUTLINGER = 10
    PR_TUNABLE_TIMEOUTLOGIN = 300
    PR_TUNABLE_TIMEOUTNOXFER = 300
    PR_TUNABLE_TIMEOUTSTALLED = 3600
    PR_TUNABLE_XFER_SCOREBOARD_UPDATES = 10
```

**Comment by Castaglia** [comment]
As far as I know, there haven't been many reports of POSIX ACL usage, good or bad.  Given that, and given that last log message of:
```
2024-03-19 17:24:44,933 [12] <fsio:8>: using system llistxattr() for path '/'
```
would it be possible to try adding this to your configuration, effectively disabling the `mod_facl` module, as an experiment?
```
<IfModule mod_facl.c>
  FACLEngine off
</IfModule>
```
If this "fixes" this out-of-memory situation, then we've narrowed down the possible code paths to examine for issues.  And if it doesn't fix that, we've ruled out some possibilities.

**Comment by cfiehe** [comment]
Hi @Castaglia,
sorry for still causing trouble... Nice catch 😃. I have added
```
<Global>
    <IfModule mod_core.c>
        FSOptions IgnoreExtendedAttributes
    </IfModule>
</Global>
```
and the mount succeeds 👍. I actually do not understand why, but that is a step into the right direction.

Can I provide more information?

**Comment by cfiehe** [comment]
Because this behavior seems to be related with the underlying filesystem, I must admit that all user data gets placed on a NFSv4 share. I do not know, if there is a correlation.

**Comment by Castaglia** [comment]
Perhaps.  My next step is to attempt to reproduce this behavior locally, using Ubuntu 22.04.4, to see if it's related to that OS version, or if it's something else.

Are you building ProFTPD from source?  If so, it'd be helpful if you could attach the `config.h` file, which is generated by running the `./configure ...` command.  That generated file contains the results of the Autoconf tests for extended attribute support (headers, specific libraries/functions, etc), which also play in specifically which function is being called for these extended attribute issues.

**Comment by cfiehe** [comment]
I am building ProFTPD from source within a multi-stage Dockerfile. The destination image is a default Ubuntu 22.04. I tried to be as close to the Ubuntu packaged version. I have copied the `config.h` from the container and attached it here.
[config.h.txt](https://github.com/proftpd/proftpd/files/14703707/config.h.txt)

When you need more input, I will try to gather the requested information.

**Comment by cfiehe** [comment]
This is my Dockerfile:
```
FROM harbor-registry.mycompany.com/docker_hub/ubuntu:22.04 as base

ARG PROFTPD_NAME=proftpd
ARG PROFTPD_CONFIG_DIR="/etc/${PROFTPD_NAME}"
ARG PROFTPD_OPTS=" --config /etc/${PROFTPD_NAME}/proftpd.conf"

# Install packages
ENV DEBIAN_FRONTEND noninteractive
RUN apt update && \
    apt install -y \
        apt-transport-https \
        apt-utils \
        curl \
        gnupg2 \
        software-properties-common && \
    apt install -y \
        acl \
        bash \
        ca-certificates \
        dnsutils \
        iproute2 \
        locales \
        lsb-release \
        net-tools \
        sudo \
        vim && \
    apt full-upgrade -y && \
    apt clean && \
    apt autoremove -y

#------------------------------------------------------------------------------------------

FROM base as build

ARG PROFTPD_VERSION
ARG PROFTPD_MOD_CLAMAV_VERSION
ARG PROFTPD_MOD_PROXY_PROTOCOL_VERSION
ARG PROFTPD_MOD_VROOT_VERSION

ARG PROFTPD_MODULES="\
mod_auth_otp:\
mod_ban:\
mod_clamav:\
mod_copy:\
mod_ctrls_admin:\
mod_deflate:\
mod_digest:\
mod_dnsbl:\
mod_dynmasq:\
mod_exec:\
mod_facl:\
mod_geoip:\
mod_ident:\
mod_ifsession:\
mod_ifversion:\
mod_ldap:\
mod_load:\
mod_log_forensic:\
mod_proxy_protocol:\
mod_qos:\
mod_quotatab:\
mod_quotatab_file:\
mod_quotatab_ldap:\
mod_quotatab_radius:\
mod_quotatab_sql:\
mod_radius:\
mod_ratio:\
mod_readme:\
mod_redis:\
mod_rewrite:\
mod_sftp:\
mod_sftp_pam:\
mod_sftp_sql:\
mod_shaper:\
mod_site_misc:\
mod_snmp:\
mod_sql:\
mod_sql_mysql:\
mod_sql_odbc:\
mod_sql_passwd:\
mod_sql_postgres:\
mod_statcache:\
mod_tls:\
mod_tls_fscache:\
mod_tls_redis:\
mod_tls_shmcache:\
mod_unique_id:\
mod_vroot:\
mod_wrap:\
mod_wrap2:\
mod_wrap2_file:\
mod_wrap2_redis:\
mod_wrap2_sql"

# Install packages
RUN apt update && \
    apt install -y \
        autoconf \
        build-essential \
        gcc \
        git \
        libacl1-dev \
        libcap-dev \
        libgeoip-dev \
        libhiredis-dev \
        libidn2-dev \
        libldap2-dev \
        libmd-dev \
        libmysqlclient-dev \
        libncurses-dev \
        libpam-dev \
        libpcre2-dev \
        libpq-dev \
        libsasl2-dev \
        libsodium-dev \
        libssl-dev \
        libwrap0-dev \
        make \
        unixodbc-dev \
        zlib1g-dev && \
    apt full-upgrade -y && \
    apt clean && \
    apt autoremove -y

COPY patches/ /proftpd-patches/

ADD apply-patches.sh /
RUN chmod a+x /apply-patches.sh

# Compile ProFTPD
RUN git clone https://github.com/proftpd/proftpd.git proftpd && \
    (cd proftpd && git checkout ${PROFTPD_VERSION}) && \
    git clone https://github.com/jbenden/mod_clamav.git proftpd-mod_clamav && \
    (cd proftpd-mod_clamav && git checkout ${PROFTPD_MOD_CLAMAV_VERSION}) && \
    git clone https://github.com/Castaglia/proftpd-mod_proxy_protocol.git proftpd-mod_proxy_protocol && \
    (cd proftpd-mod_proxy_protocol && git checkout ${PROFTPD_MOD_PROXY_PROTOCOL_VERSION}) && \
    git clone https://github.com/Castaglia/proftpd-mod_vroot.git proftpd-mod_vroot && \
    (cd proftpd-mod_vroot && git checkout ${PROFTPD_MOD_VROOT_VERSION}) && \
    mv proftpd-mod_clamav/*.h proftpd-mod_clamav/*.c proftpd/contrib && \
    mv proftpd-mod_proxy_protocol/*.c proftpd/contrib && \
    mv proftpd-mod_vroot proftpd/contrib/mod_vroot && \
    ./apply-patches.sh && \
    (cd proftpd && \
    ./configure CFLAGS="-DBAN_LIST_MAXSZ=1024"\
        --prefix=/usr \
        --mandir=/usr/share/man \
        --sysconfdir=/etc/${PROFTPD_NAME} \
        --localstatedir=/run \
        --libexecdir=/usr/lib/${PROFTPD_NAME} \
        --with-includes=/usr/include/mysql:/usr/include/postgresql \
        --with-shared=${PROFTPD_MODULES} \
        --enable-autoshadow \
        --enable-ctrls \
        --enable-dso \
        --enable-facl \
        --enable-ipv6 \
        --enable-openssl \
        --enable-pcre2 \
        --enable-nls \
        --enable-redis \
        --enable-sendfile \
        --with-lastlog=/var/log/lastlog \
        --disable-strip \
        --disable-ident && \
    make all && \
    make install DESTDIR=/proftpd-build) && \
    cp -p proftpd/contrib/xferstats.holger-preiss proftpd-build/usr/sbin/ftpstats

#------------------------------------------------------------------------------------------

FROM base

LABEL org.opencontainers.image.created=$BUILD_DATE
LABEL org.opencontainers.image.description="A ProFTPD Docker container based on Ubuntu 22.04 LTS (Jammy Jellyfish)."
LABEL org.opencontainers.image.title="proftpd"
LABEL org.opencontainers.image.version=$BUILD_VERSION

# Install packages
RUN apt update && \
    apt install -y \
        libacl1 \
        libc6 \
        libcap2 \
        libcrypt1 \
        libgeoip1 \
        libhiredis0.14 \
        libidn2-0 \ 
        libldap-2.5-0 \
        libmysqlclient21 \
        libncursesw6 \
        libodbc2 \
        libpam-runtime \
        libpam0g \
        libpcre2-8-0 \
        libpcre2-posix3 \
        libpq5 \
        libsodium23 \
        libssl3 \ 
        libtinfo6 \
        libwrap0 \
        netbase \
        openssh-client \
        perl-modules-5.34 \
        sssd \
        tzdata \
        ucf \
        zlib1g && \
    apt full-upgrade -y && \
    apt clean && \
    apt autoremove -y && \
    rm -rf /tmp/* && \
    rm -rf /var/lib/apt/lists/* && \
    rm -rf /var/log/* && \
    rm -rf /var/tmp/*

COPY --from=build /proftpd-build /

ADD files/. /

RUN groupadd proftpd && \
    useradd -g proftpd proftpd && \
    mkdir /var/log/proftpd && \
    chmod a+x /docker-entrypoint.sh

VOLUME /srv/ftp

EXPOSE 21 22 49152-49407

ENTRYPOINT ["/docker-entrypoint.sh"]

CMD ["/usr/sbin/proftpd", "--nodaemon", "--quiet"]
```

**Comment by cfiehe** [comment]
I was able to get xattr working, running directly into https://github.com/xattr/xattr/issues/90.

I get the following for the user's home directory:
```
root@a754be577990:/# xattr /srv/ftp/myuser/
system.nfs4_acl
root@a754be577990:/# xattr -l /srv/ftp/myuser/
system.nfs4_acl:
0000   00 00 00 03 00 00 00 00 00 00 00 00 00 16 01 E7    ...............ç
0010   00 00 00 06 4F 57 4E 45 52 40 00 00 00 00 00 00    ....OWNER@......
0020   00 00 00 00 00 12 00 80 00 00 00 06 47 52 4F 55    ............GROU
0030   50 40 00 00 00 00 00 00 00 00 00 00 00 12 00 80    P@..............
0040   00 00 00 09 45 56 45 52 59 4F 4E 45 40 00 00 00    ....EVERYONE@...
```

**Comment by Castaglia** [comment]
That helps confirm a suspicion I had earlier today, when reading the code in question, specifically here:
* https://github.com/proftpd/proftpd/blob/master/src/fsio.c#L718

I don't think we're currently handling the case where `listxattr(2)` returns 0, indicating "no extended attributes found for this path", well.  We handle an error case (`res < 0`), otherwise we assume that the value is the amount of memory to allocate.  Allocating zero memory, and working with it, is "here be dragons" territory.

I should have a PR for experimentally validating this hypothesis a little later today, hopefully.

**Comment by Castaglia** [comment]
Can you verify whether https://github.com/proftpd/proftpd/pull/1786 addresses this issue, without need for the `FSOptions IgnoreExtendedAttributes` workaround?

**Comment by cfiehe** [comment]
I built the new version and removed the setting `FSOptions IgnoreExtendedAttributes` from the configuration. The out-of-memory exception does **not** occur. The issue seems to be fixed 😍. Thank you very much!

**Comment by Castaglia** [comment]
This is now merged/fixed in master, and backported to the 1.3.8 branch.  Thanks!
